### PR TITLE
[Dashboard] Add a default entrypoint for all jobs

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

NOTE: Ray client still doesn't work. We should do this as a follow-up. 

This PR does
- Add the (optional. Java & Cpp doesn't work) entrypoint information to all jobs
- Add the corresponding information to the dashboard. (and I fixed the issue long driver names will screw up the table UI).

Sample screenshot
Screen Shot 2022-11-06 at 11 18 11 PM

The current entrypoint detect logic works well for regular script, Python, Ipython, and Jupyter. Unforutnately, adding a test for Jupyter was pretty hard, so I will just attach how this will look from the notebook below.
Screen Shot 2022-11-06 at 10 50 44 PM

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -1894,6 +1896,8 @@ def connect(
             it during startup as a command line argument.
         ray_debugger_external: If True, make the debugger external to the
             node this worker is running on.
+        entrypoint: The name of the entrypoint script. Ignored unless the
+            mode != SCRIPT_MODE
```
Comment (rickyyx) [inline python/ray/_private/worker.py:1900]:
```suggestion
        entrypoint: The name of the entrypoint script. Empty if the
            mode != SCRIPT_MODE
```

File: python/requirements_test.txt [hunk 2]
```diff
@@ -68,6 +68,7 @@ testfixtures==7.0.0
 werkzeug==2.1.2
 xlrd==2.0.1
 memray; platform_system != "Windows" and sys_platform != "darwin"
+jupyter
```
Comment (rickyyx) [inline python/requirements_test.txt:71]: should we be pinning version here, cc @krfricke

**Review by rickyyx — APPROVED** [review]

File: python/requirements_test.txt (hunk 2 above)
Comment (rkooo567) [reply python/requirements_test.txt:71]: actually we don't need this anymore. I will remove it..

**Comment by ericl** [comment]
Seems there is a merge conflict.

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**